datafusion/src/optimizer/simplify_expressions.rs

```diff
@@ -785,180 +786,166 @@ mod tests {
+        let expected = col("c2");
 
         assert_eq!(simplify(&expr), expected);
-        Ok(())
     }
 
     #[test]
-    fn test_simplify_divide_by_same() -> Result<()> {
-        let expr = binary_expr(col("c"), Operator::Divide, col("c"));
+    fn test_simplify_divide_by_same() {
+        let expr = binary_expr(col("c2"), Operator::Divide, col("c2"));
         let expected = lit(1);
 
         assert_eq!(simplify(&expr), expected);
-        Ok(())
     }
 
     #[test]
-    fn test_simplify_simple_and() -> Result<()> {
+    fn test_simplify_simple_and() {
         // (c > 5) AND (c > 5)
-        let expr = (col("c").gt(lit(5))).and(col("c").gt(lit(5)));
-        let expected = col("c").gt(lit(5));
+        let expr = (col("c2").gt(lit(5))).and(col("c2").gt(lit(5)));
+        let expected = col("c2").gt(lit(5));
 
         assert_eq!(simplify(&expr), expected);
-        Ok(())
     }
 
     #[test]
-    fn test_simplify_composed_and() -> Result<()> {
+    fn test_simplify_composed_and() {
         // ((c > 5) AND (d < 6)) AND (c > 5)
         let expr = binary_expr(
-            binary_expr(col("c").gt(lit(5)), Operator::And, col("d").lt(lit(6))),
+            binary_expr(col("c2").gt(lit(5)), Operator::And, col("d").lt(lit(6))),
             Operator::And,
-            col("c").gt(lit(5)),
+            col("c2").gt(lit(5)),
         );
         let expected =
-            binary_expr(col("c").gt(lit(5)), Operator::And, col("d").lt(lit(6)));
+            binary_expr(col("c2").gt(lit(5)), Operator::And, col("d").lt(lit(6)));
 
         assert_eq!(simplify(&expr), expected);
-        Ok(())
     }
 
     #[test]
-    fn test_simplify_negated_and() -> Result<()> {
+    fn test_simplify_negated_and() {
         // (c > 5) AND !(c > 5) -- can't remove
         let expr = binary_expr(
-            col("c").gt(lit(5)),
+            col("c2").gt(lit(5)),
             Operator::And,
-            Expr::not(col("c").gt(lit(5))),
+            Expr::not(col("c2").gt(lit(5))),
         );
         let expected = expr.clone();
 
         assert_eq!(simplify(&expr), expected);
-        Ok(())
     }
 
     #[test]
-    fn test_simplify_or_and() -> Result<()> {
+    fn test_simplify_or_and() {
         // (c > 5) OR ((d < 6) AND (c > 5) -- can remove
         let expr = binary_expr(
-            col("c").gt(lit(5)),
+            col("c2").gt(lit(5)),
             Operator::Or,
-            binary_expr(col("d").lt(lit(6)), Operator::And, col("c").gt(lit(5))),
+            binary_expr(col("d").lt(lit(6)), Operator::And, col("c2").gt(lit(5))),
         );
-        let expected = col("c").gt(lit(5));
+        let expected = col("c2").gt(lit(5));
 
         assert_eq!(simplify(&expr), expected);
-        Ok(())
     }
 
     #[test]
-    fn test_simplify_and_and_false() -> Result<()> {
-        let expr =
-            binary_expr(lit(ScalarValue::Boolean(None)), Operator::And, lit(false));
+    fn test_simplify_and_and_false() {
```

Comment:
Maybe it should be `test_simplify_null_and_false`?

Reply:
Good 👀  👍